--- nginx-0.7.61/auto/os/linux.old	2009-06-28 16:31:31.000000000 +0200
+++ nginx-0.7.61/auto/os/linux	2009-06-28 16:32:08.000000000 +0200
@@ -18,7 +18,7 @@ CC_AUX_FLAGS="$cc_aux_flags -D_GNU_SOURC
 # Linux kernel version
 
 version=$((`uname -r \
-         | sed 's/^\([^.]*\)\.\([^.]*\)\.\([^.-]*\).*/\1*256*256+\2*256+\3/'`))
+         | sed 's/^\([^.]*\)\.\([^.]*\)\.\([^._]*\).*/\1*256*256+\2*256+\3/'`))
 
 version=${version:-0}
 
